onedal/covariance/incremental_covariance.py

```diff
@@ -58,10 +58,49 @@ def __init__(self, method="dense", bias=False, assume_centered=False):
         self._reset()
 
     def _reset(self):
+        self._need_to_finalize = False
         self._partial_result = self._get_backend(
             "covariance", None, "partial_compute_result"
         )
 
+    def __getstate__(self):
+        """
+        Converts estimator's data to serializable format.
+        All tables contained in partial result are converted to np.arrays.
+
+        Notes
+        -----
+        Since finalize_fit can't be dispatched without directly provided queue
+        and the dispatching policy can't be serialized, the computation is finalized
+        here and the policy is not saved in serialized data.
+        """
+        self.finalize_fit()
+        data = self.__dict__.copy()
+
+        tables_to_save = ["partial_n_rows", "partial_crossproduct", "partial_sums"]
+        partial_result_data = {
+            table_name: from_table(getattr(data["_partial_result"], table_name))
+            for table_name in tables_to_save
+        }
+        data["_partial_result"] = partial_result_data
+        data.pop("_policy", None)
+
+        return data
+
+    def __setstate__(self, data):
+        """
```

Comment:
Just so I understand, what happens when the _policy attribute doesn't exist?  What does that default to in the computation, or will it get auto-generated with a _get_policy call somewhere later on (after unpickling)?

Reply:
yes, it will be autogenerated using provided data. because we finalized computation before serialization, the case when we have to call finalize_fit without data is not possible.